server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -383,6 +384,13 @@ protected void doXContentBody(XContentBuilder builder, Params params) throws IOE
         return indexAnalyzers;
     }
 
+    /**
+     * @return a PostingsFormat for this field, or {@code null} if the default postings should be used
+     */
+    public PostingsFormat postingsFormat() {
```

Comment:
I don't think we should reopen this capability for plugins. It was removed in https://github.com/elastic/elasticsearch/issues/8746 and I don't think that the policy should change. 
Can we make this change without exposing custom postings format in field mappers ?

Reply:
I've reworked things so that the MappingLookup detects completion mappers at build time and will return the relevant per-field postings format, removing the pluggability again.